File: CrouskieBackOffice/src/crouskiebackoffice/model/Color.java
package crouskiebackoffice.model;

/**
 * Classe représentant une couleur.
 *
 * @author Nom de l'auteur
 */
public class Color implements HasName {

    /**
     * Nom de la couleur.
     */
    private final String name;
    /**
     * Code de la couleur, pour la représenté.
     */
    private final String code;

    /**
     * Constructeur de la couleur.
     *
     * @param name Nom de la couleur.
     */
    public Color(String name) {
        this.name = name;
        this.code = null;
    }

    /**
     * Constructeur de la couleur.
     *
     * @param name Nom de la couleur.
     * @param code Code represantant la couleur
     */
    public Color(String name, String code) {
        this.name = name;
        this.code = code;
    }

    /**
     * Retourne le nom de la couleur.
     *
     * @return Le nom de la couleur.
     */
    @Override
    public String getName() {
        return name;
    }

    /**
     * Retourne le code de la couleur.
     *
     * @return Le code de la couleur
     */
    public String getCode() {
        return code;
    }

    /**
     * Retourne le nom de la couleur.
     *
     * @return Le nom de la couleur.
     */
    @Override
    public String toString() {
        return name;
    }
}

File: entities/Color.php
<?php
/**
 * Cette classe représente une couleur d'un produit
 */
class Color
{
    // ----------- Properties -------------

    private string $_name;
    private string $_code;

    /**
     * Constructeur de la classe Color
     * @param string $name le nom de la couleur
     * @param string $code le code informatique de la couleur (hex "#ffffff" ou bien "rgb(0,0,0)")
     */
    function __construct(string $name, string $code)
    {
        $this->_name = $name;
        $this->_code = $code;
    }

    // ----------- Getters -------------

    /**
     * Retourne le nom de la couleur
     * @return string le nom de la couleur
     */
    public function getName() : string {
        return $this->_name;
    }

    /**
     * Retourn le code de la couleur
     * @return string Le code de la couleur
     */
    public function getCode() : string {
        return $this->_code;
    }
}
